HPCC-19632 CEsdlSDSStore should honor Dali attach state

mayx opened the pull request:
- Added methods and variable to handle attach state
- Check if attached before calling querySDS()

Signed-off-by: mayx <yanrui.ma@lexisnexisrisk.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 59bb1ca38e6e5a6bb140a07de3aa05aec4482a61
Build: success
Build: success
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |808 |808 |0 |
|test (thor) |732 |732 |0 |
|test (roxie) |882 |882 |0 |


HPCC Stop: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   36 sec (00:00:36) |  185 sec (00:03:05) |    0 sec (00:00:00) |    2 sec (00:00:02) |   18 sec (00:00:18) | 1489 sec (00:24:49) |   19 sec (00:00:19) | 1749 sec (00:29:09) |

mayx committed:
HPCC-19632 CEsdlSDSStore should honor Dali attach state

- Added methods and variable to handle attach state
- Check if attached before calling querySDS()

Signed-off-by: mayx <yanrui.ma@lexisnexisrisk.com>

rpastrana commented on esp/services/esdl_svc_engine/esdl_store.cpp:
Minor, but most conventions use lower case initial letter on method name

rpastrana reviewed (commented):
@mayx just one minor comment

mayx commented on esp/services/esdl_svc_engine/esdl_store.cpp:
Changed method name to checkQuerySDS(), please review again @rpastrana

rpastrana reviewed (approved):
@mayx approved

HPCCSmoketest commented:
Process of PR-12036, label: HPCC-19632-AttachState starts now. Estimated completion time is ~ 0.5 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.4.1708 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 5e6990043237f72480291000dbcb8746f55316c1
Build: success
Build: success
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |808 |808 |0 |
|test (thor) |732 |732 |0 |
|test (roxie) |882 |882 |0 |


HPCC Stop: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   33 sec (00:00:33) |  187 sec (00:03:07) |    0 sec (00:00:00) |    2 sec (00:00:02) |   18 sec (00:00:18) | 1510 sec (00:25:10) |   20 sec (00:00:20) | 1770 sec (00:29:30) |
